include/kodlab_mjbots_sdk/limb_base.h

```diff
@@ -0,0 +1,155 @@
+ *
+ */
+
+#pragma once
+#include <limits>
+#include <memory>
+#include <vector>
+#include <type_traits>
+#include <string>
+#include "kodlab_mjbots_sdk/joint_base.h"
+#include <Eigen/Dense>
+
+namespace kodlab
+{
+    /**
+     * Abstract limb class that takes in a series of joints and treats it as an abstract limb
+     * Contains the kinematics and configurations for a leg
+     * @brief Abstract base class for legs consisting of multiple joints to be used in the robot class
+     */
+
+    struct LimbConfig
+    {
+        std::vector<std::array<float, 3>> actuator_offsets; // Offset between centers of two joints, in the frame of the first joint
+        std::vector<Eigen::Matrix3f> actuator_orientations; // Rotation matrices for each joint with respect to the previous joint (begins with R0_1)
+        Eigen::MatrixXf selection_matrix;
+    };
+
+    class LimbBase
+    {
+    public:
+        /**
+         * @brief Construct a new Limb Base object with a name
+         *
+         * @param name          /// Sets the limb name
+         * @param joints        /// Joints that make up the leg
+         * @param config        /// Configuration of the joints in the leg
+         *
+         */
+
+        LimbBase(
+            std::string name,
+            std::vector<std::shared_ptr<JointBase>> joints,
+            LimbConfig config);
+
+        /**
+         * @brief Construct a new Limb Base object
+         *
+         * @param joints        /// Joints that make up the leg
+         * @param config        /// Configuration of the joints in the leg
+         *
+         */
+
+        LimbBase(
+            std::vector<std::shared_ptr<JointBase>> joints,
+            LimbConfig config);
+
+        virtual ~LimbBase(){};
+
+        /**
+         * @brief Update position and velocity of each joint within leg
+         *
+         * @param pos_list desired positions of each joint
+         * @param vel_list desired velocities of each joint
+         *
+         */
+        void Update(std::vector<float> pos_list, std::vector<float> vel_list);
+
+        /**
+         * @brief Calculates the forward kinematics of the leg
+         *
+         * @return Eigen::VectorXf
+         */
+        virtual void ForwardKinematics() = 0;
+
+        /**
+         * @brief Calculates the Jacobian of the leg
+         *
+         * @return Eigen::MatrixXf
+         */
+        virtual void Jacobian() = 0;
+
+        /**
+         * @brief Calculates the Inverse kinematics of a leg given a desired position
+         * measured from the origin of the leg (????)
+         *
+         * @param x_des desired x position
+         * @param y_des desired y position
+         * @param z_des desired z position
+         *
+         * @return std::vector<float> positions
+         */
+
+        virtual void InverseKinematics(std::vector<float> EE_pos) = 0;
```

Comment:
Pass by constant reference per [ISO reference](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f15-prefer-simple-and-conventional-ways-of-passing-information):
```suggestion
        virtual void InverseKinematics(const std::vector<float> &EE_pos) = 0;
```

Reply:
fixed for larger formatting commit